File: app/models/champion.rb
class Champion < ActiveRecord::Base
  validates :name, presence: true

  BOOLEAN_COLUMNS = %w{
    active
    free_to_play
    ranked_play_enabled
    bot_enabled
    bot_mm_enabled
  }

  INTEGER_COLUMNS = %w{
    difficulty_rank
    attack_rank
    defense_rank
    magic_rank
  }

  INTEGER_COLUMNS.each do |column|
    validates column, numericality: { only_integer: true, allow_nil: true, allow_blank: true }
  end
end

File: app/assets/javascripts/models/champion.js
Brokenwings.Champion = DS.Model.extend({
  name: DS.attr('string'),
  active: DS.attr('boolean'),
  free_to_play: DS.attr('boolean'),
  ranked_play_enabled: DS.attr('boolean'),
  bot_enabled: DS.attr('boolean'),
  bot_mm_enabled: DS.attr('boolean'),
  difficulty_rank: DS.attr('number'),
  attack_rank: DS.attr('number'),
  defense_rank: DS.attr('number'),
  magic_rank: DS.attr('number')
});